GameEngine/GameState.cs

```diff
@@ -85,10 +93,23 @@ public GameState Clone()
             {
                 Board = Board.Clone(),
                 Deck = Deck.Clone(),
-                Hand = Hand.Clone(),
+                Hands = Hands.Select( hand => hand.Clone() ).ToList(),
                 SunCounter = SunCounter,
-                SunSpaces = SunSpaces
+                SunSpaces = SunSpaces,
+                CurrentPlayer = CurrentPlayer,
             };
         }
+
+        private void NextPlayer()
```

Comment:
I have a weak preference for adding, then doing the modular arithmetic, like
```
CurrentPlayer++;
CurrentPlayer %= Hands.Count;
```

Reply:
Why?

Reply:
Nope, forget it, I see it now.  (Although, I'm still interested in your specific perspective.)

Reply:
👍